mayday.go

```diff
@@ -2,289 +2,94 @@ package main
 
 import (
 	"archive/tar"
-	"compress/gzip"
-	"fmt"
-	"io"
+	"encoding/json"
 	"io/ioutil"
 	"log"
 	"os"
-	"os/exec"
-	"path"
 	"path/filepath"
-	"regexp"
-	"strings"
 	"time"
 
 	"github.com/coreos/mayday/mayday"
-
-	"github.com/coreos/mayday/Godeps/_workspace/src/github.com/coreos/go-systemd/dbus"
 )
 
 const (
-	dirPrefix = "mayday"
+	dirPrefix = "/mayday"
 )
 
-var (
-	files = []mayday.File{
-		{
-			Path: "/proc/vmstat",
-		},
-		{
-			Path: "/proc/meminfo",
-			Link: "meminfo",
-		},
-		{
-			Path: "/etc/os-release",
-			Link: "os-release",
-		},
-		{
-			Path: "/proc/mounts",
-			Link: "mounts",
-		},
-	}
-	commands = []mayday.Command{
-		{
-			Args: []string{"hostname"},
-			Link: "hostname",
-		},
-		{
-			Args: []string{"date"},
-			Link: "date",
-		},
-		{
-			Args: []string{"systemd-cgls"},
-		},
-		{
-			Args: []string{"systemd-cgtop", "-n1"},
-		},
-		{
-			Args: []string{"ps", "fauxwww"},
-			Link: "ps",
-		},
-		{
-			Args: []string{"lsmod"},
-			Link: "lsmod",
-		},
-		{
-			Args: []string{"lspci"},
-			Link: "lspci",
-		},
-		{
-			Args: []string{"lsof", "-b", "-M", "-n", "-l"},
-			Link: "lsof",
-		},
-		{
-			Args: []string{"blkid"},
-		},
-		{
-			Args: []string{"btrfs", "fi", "show"},
-		},
-		{
-			Args: []string{"df", "-al"},
-			Link: "df",
-		},
-		{
-			Args: []string{"ip", "-o", "addr", "show"},
-			Link: "ip_addr_show",
-		},
-		{
-			Args: []string{"ip", "-o", "link", "show"},
-			Link: "ip_link_show",
-		},
-		{
-			Args: []string{"ip", "-o", "route", "show"},
-			Link: "ip_route_show",
-		},
-		{
-			Args: []string{"netstat", "-neopa"},
-			Link: "netstat",
-		},
-		{
-			Args: []string{"df", "-ali"},
-		},
-		{
-			Args: []string{"free", "-m"},
-			Link: "free",
-		},
-		{
-			Args: []string{"systemctl", "list-units", "-a"},
-			Link: "all_units",
-		},
-		{
-			Args: []string{"systemctl", "list-units", "--state=running"},
-			Link: "running_units",
-		},
-		{
-			Args: []string{"systemctl", "status", "etcd.service"},
-			Link: "etcd_status",
-		},
-		{
-			Args: []string{"systemctl", "status", "etcd2.service"},
-			Link: "etcd2_status",
-		},
-		{
-			Args: []string{"systemctl", "status", "fleet.service"},
-			Link: "fleet_status",
-		},
-		{
-			Args: []string{"systemctl", "status", "flanneld.service"},
-			Link: "flanneld_status",
-		},
-	}
-)
+type Config struct {
+	Files    []mayday.File
+	Commands []mayday.Command
+}
 
-func getJournals(dir string) error {
-	c, err := dbus.New()
+func readConfig() ([]mayday.File, []mayday.Command, error) {
```

Comment:
I agree with @colhom in spirit on this, but if I were going to make this testable, I'd change it to parseConfig, and just accept a string as the argument, and do the config reading in main. 

I think it'd be best if this returned a config struct though, rather than just using the config struct temporarily to parse the json.